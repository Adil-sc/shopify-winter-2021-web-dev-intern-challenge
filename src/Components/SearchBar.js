import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSearch } from '@fortawesome/free-solid-svg-icons'

const SearchBar = (props) => {
  return (
    <div className="h-auto  bg-white shadow rounded-md">
      <div className="p-4">
        <p className="text-sm pb-1 text-gray-700">Movie Title</p>

        <div className="flex items-center">
          <span className="absolute pl-3 ">
            <FontAwesomeIcon icon={faSearch} />
          </span>
          <input
            className="shadow appearance-none border px-10 rounded w-full p-2 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
            id="username"
            type="text"
            placeholder="Search movies..."
            onChange={(e) => props.onSearchChange(e.target.value.trim())}
          ></input>
        </div>
      </div>
    </div>
  )
}

export default SearchBar
