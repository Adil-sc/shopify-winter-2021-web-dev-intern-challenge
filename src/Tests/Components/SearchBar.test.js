import React from 'react'
import { shallow } from 'enzyme'
import SearchBar from '../../Components/SearchBar'

describe('<SearchBar />', () => {
  it('should render SearchBar correctly', () => {
    const wrapper = shallow(<SearchBar />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should display input bar', () => {
    const wrapper = shallow(<SearchBar />)
    expect(wrapper.find('input').text()).toBe('')
  })
})
