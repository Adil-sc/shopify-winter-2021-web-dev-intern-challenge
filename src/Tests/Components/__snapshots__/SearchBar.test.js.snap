// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should render SearchBar correctly 1`] = `
<div
  className="h-auto  bg-white shadow rounded-md"
>
  <div
    className="p-4"
  >
    <p
      className="text-sm pb-1 text-gray-700"
    >
      Movie Title
    </p>
    <div
      className="flex items-center"
    >
      <span
        className="absolute pl-3 "
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f002",
                "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
              ],
              "iconName": "search",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
      <input
        className="shadow appearance-none border px-10 rounded w-full p-2 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="username"
        onChange={[Function]}
        placeholder="Search movies..."
        type="text"
      />
    </div>
  </div>
</div>
`;
